aodndata/acorn/handler.py

```diff
@@ -48,22 +46,55 @@ def __init__(self, *args, **kwargs):
         super(AcornHandler, self).__init__(*args, **kwargs)
         self.allowed_extensions = ['.nc']
 
+    def preprocess(self):
+        nc_collection = self.file_collection[0]
+
+        # if file contains any of FILE_TYPE_NEED_INDEX, we index, otherwise publish only
+        if any(s in get_type(nc_collection._name) for s in FILE_TYPE_NEED_INDEX):
+            nc_collection.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+        else:
+            nc_collection.publish_type = PipelineFilePublishType.UPLOAD_ONLY
+
+        # check if file with same dest path already on s3. If yes, check it's date_created nc attribute to know
+        # if we need to overwrite this object or not
+            destination_s3 = self.dest_path(nc_collection._name)
+
+        #nc_collection.dest_path = os.path.join(destination_s3, nc_collection.name)
+        #storage_query_res = self.state_query.query_storage(destination_s3)
+        #if storage_query_res:
+        #    creation_date_nc_pipeline = self.get_creation_date(nc_collection)  # creation date of the file in the pipeline
+
+        #    #TODO get leigh to create the funcion below
+        #    nc_s3_path = download_s3(destination_s3)
+        #    creation_date_nc_s3 = get_creation_date(nc_s3_path)  # creation date of the file on s3
+
+        #    if creation_date_nc_pipeline > creation_date_nc_s3:
+        #        nc_collection.publish_type = PipelineFilePublishType.NO_ACTION
+
+    @staticmethod
+    def get_creation_date(netcdf_path):
+        with Dataset(netcdf_path, mode='r') as nc_obj:
+            nc_pipeline_date_created_str = nc_obj.date_created
+            nc_pipeline_date_created = datetime.strptime(nc_pipeline_date_created_str, '%Y-%m-%dT%H:%M:%SZ')
+        return nc_pipeline_date_created
+
     @staticmethod
     def dest_path(filepath):
-        m = re.search(r'^IMOS_ACORN_([A-Z].*)_([0-9]{8}T[0-9]{6}Z)_([A-Z].*)_(FV0[0-2])_(.*)\.nc$', filepath)
+        m = re.search(r'^IMOS_ACORN_([A-Z].*)_([0-9]{8}T[0-9]{6}Z)_([A-Z]{3,4})_(FV0[0-1])_(.*)\.nc$',
```

Comment:
Just like in the original regex:
```bash
ACORN_RT_REGEX='^IMOS_ACORN_[[:alpha:]]{1,2}_[[:digit:]]{8}T[[:digit:]]{6}Z_[[:alpha:]]{3,4}_FV00_(radial|sea-state|wavespec|windp|wavep|1-hour-avg)\.nc$'
```
Can we please continue to restrict on the last bit on the file name before the extension? -> (radial|sea-state|wavespec|windp|wavep|1-hour-avg)